runtime/onert/core/include/ir/train/TrainableGraph.h

```diff
@@ -101,6 +101,7 @@ class TrainableGraph : public IGraph
   void changeDerivativeShape(const OperandIndex &ind, const ir::Shape &new_shape);
   void addInput(const OperandIndex &ind, const std::string &name = "");
   void addOutput(const OperandIndex &ind, const std::string &name = "");
+  void addLoss(const IOIndex &pred_io_ind, const OperandIndex &loss_ind);
```

Comment:
- What does `addLoss`?
- What is `pred_io_ind` and `loss_ind`?

 I am trying to figure out by reading`LossInsertionPass` from #11035.

Reply:
The `addLoss` inserts a pair information about `pred_io_ind` and `loss_ind`.
`LossInsertionPass` inserts new Loss operation for calculating loss value using prediction value and expected value. User can invoke the `nnfw_train_get_loss` function to obtain loss value about each output index. Currently there is no way to get the index of the newly added Loss Op for output index. This map has information about the index of Loss Op for a given output index.

Reply:
@jyoungyun Thank you for explanation.

Here is my understanding.

- `addLoss` adds a key-value pair into unordered_map (`pred_io_ind`<sup>IOIndex</sup> → `loss_ind` <sup>OperandIndex</sup>).
- `pred_io_ind` is simply the output index. (Currently and simple case, it is 0.)
- `loss_ind` is the _output_ operand of `Loss` operation.
- `addLoss` will be called as many times as the number of model output.

Reply:
(optional)

I would like to make `addLoss` similar to `addInput` and `addOutput`.

```cc
void addInput(const OperandIndex &ind, const std::string &name = "");
void addOutput(const OperandIndex &ind, const std::string &name = "");
```

`addInput` and `addOutput` inform `TrainableGraph`
- 1st argument (OperandIndex) is input (or output)
- 2nd argument gives additional information about the OperandIndex


```suggestion
  void addLoss(const OperandIndex &loss_ind, const IOIndex &pred_io_ind);
```

It will inform `TrainableGraph`
- 1st argument (OperandIndex) is loss.
- 2nd argument gives additional information that IOIndex is associated with the operand.

Reply:
I applied your suggestion. Please review this PR again.